File: backend/internal/core/model/response.go
package model

import (
	"github.com/google/uuid"
	"time"
)

type CreateResponseParams struct {
	QuestionID uuid.UUID
	Body       string
	ImageURLs  []string
}

type EditResponseParams struct {
	ResponseID uuid.UUID
	Body       string
}

// Response model
type Response struct {
	ID         uuid.UUID `json:"id"`
	QuestionID uuid.UUID `json:"question_id"`
	Author     User      `json:"author"`
	Body       string    `json:"body"`
	IsOwned    bool      `json:"is_owned"`
	ImageURLs  []string  `json:"image_urls"`
	CreatedAt  time.Time `json:"created_at"`
	EditedAt   time.Time `json:"edited_at"`
}

File: frontend/models/response.ts
import { User } from "@/models/user";

export type Response = {
    id: string;
    question_id: string;
    author: User;
    body: string;
    is_owned: boolean;
    image_urls?: string[]
    created_at: string;
    edited_at: string;
}

// DTOs

// CREATE RESPONSE
export type CreateResponseReq = {
    question_id: string;
    body: string;
    image_urls?: string[];
}

export type CreateResponseRes = {
    response: Response;
}

// EDIT RESPONSE
export type EditResponseReq = {
    response_id: string;
    question_id: string; // THIS IS ONLY USED BY TANSTACK
    body: string;
}

export type EditResponseRes = {
    response: Response;
}

// GET RESPONSES BY QUESTION ID
export type GetResponsesByQuestionIdRes = {
    responses: Response[];
}

// GET RESPONSE BY ID
export type GetResponseByIdRes = {
    response: Response;
}

// GET QUESTION SUMMARY
export type GetQuestionSummaryRes = {
    summary: string;
}